### ISSUE TITLE
Should there be a length limit on titles?

### ISSUE THREAD

**Issue by kmccurley** [IACR/latex#180]
Springer will sometimes split a title into title/subtitle if the title is too long. I noticed a paper today with a long title, and in iacrcc.cls it looks like this:
![image](https://github.com/IACR/latex/assets/14105363/4c76a868-4b85-4939-a9ae-882ca6d12d7e)

I don't consider it to be fatal, but it's pretty ugly.

**Comment by jwbos** [comment]
Isn't this why we have the `running` option to the `\title` macro?

**Comment by kmccurley** [comment]
While converting several papers from Crypto, I found this to be a common problem. We could presumably use xstring `\StrLen` to determine if the title is too long and at least give a warning so the copy editor could find it. Something seems wrong in the header though, because this example should fit:
![image](https://github.com/IACR/latex/assets/14105363/d3325891-9f80-4c69-9ed3-b242879179a8)
Is this because the fancyhdr dedicates equal space to L and R? We might be able to fix this by adjusting `\headwidth` to 80% of `\linewidth`

**Comment by kmccurley** [comment]
I discovered another problem with the header in crypto paper #265
![image](https://github.com/IACR/latex/assets/14105363/bf2bb7af-5e39-4699-b76d-16db269fc70b)
Notice how the title overlaps the page number on the first even page.

**Comment by jwbos** [comment]
I suggest we indeed check the length of the title and throw an error if it is too long (and suggest the authors use the running title option). It seems the maximum length which looks fine is around 75 characters but this can go wrong in a lof of ways (if special characters or math are used). Maybe suggest the authors to provide one when the title is over 65 characters?

**Comment by kmccurley** [comment]
I think you're right that we can only issue a warning. As you say, titles may contain mathematics, and it's hard to correlate the actual length of the title exactly to the number of characters. Something like $\epsilon$ turns out to be a single character but needs to be written out as \epsilon. 

When the title is long enough to wrap on a second line, we see the following in the latex log:
```
Package fancyhdr Warning: \headheight is too small (12.0pt):
```
which means that it's overflowing the head:
![image](https://github.com/IACR/latex/assets/14105363/4423c67f-c97a-4f0e-bcab-2265e77bd246)
(the extra 12pts is for the second line). The latex-submit system catches this as a warning to the author, and the copy editor should also notice this:
![image](https://github.com/IACR/latex/assets/14105363/3ab43a88-18f1-4198-99ef-aee916dd79e9)
Authors may not realize what is causing this, but the copy editor should. We can try to escalate this to an error that needs fixing by the author, but it's hard to anticipate what warnings `fancyhdr` would generate. For example, I'm not sure what happens in the log when the title doesn't wrap but overlaps the page number. That is at least visually wrong so the author might notice it.

**Comment by jwbos** [comment]
See my pull request to fix this (and especially 9ce8c8f474af16dc527d8da1520a88bee129695b).
